python_modules/dagster/dagster/_cli/workspace/cli_target.py

```diff
@@ -278,12 +287,88 @@ def get_workspace_from_kwargs(
     version: str,
     kwargs: ClickArgMapping,
 ) -> Iterator[WorkspaceRequestContext]:
+    logger.debug("Loading workspace with gRPC server")
     with get_workspace_process_context_from_kwargs(
         instance, version, read_only=False, kwargs=kwargs
     ) as workspace_process_context:
         yield workspace_process_context.create_request_context()
 
 
+def _does_origin_executable_match(origin: CodeLocationOrigin) -> bool:
+    return (
+        origin.loadable_target_origin.executable_path is None
+        or origin.loadable_target_origin.executable_path == sys.executable
+    )
+
+
+@contextmanager
+def get_auto_determined_workspace_from_kwargs(
```

Comment:
If we have this, do we even need to give the users the `--load-with-grpc` option? IIUC we can just detect which method to use and fall back to grpc when necessary. If we do it this way then maybe we can eliminate it entirely later.


Just trying to avoid adding API surface?

Reply:
Yeah, that's fair. I figured it's sort of nice to have the flag to force consistently using gRPC if for some reason a user wants to make sure the command does the same thing each time, but don't feel strongly.

Reply:
IMO the method used to load the definitions should be treated as an implementation detail. If there are errors that would somehow only be found from gRPC startup, I would think that those errors are outside the scope of the commands expected output.

Maybe I'm wrong though, will let others weigh in.